### PR TITLE
Add AmbryServerRequests support for BlobIndexAdminRequest

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
Add AmbryServerRequests support for BlobIndexAdminRequest
Add corresponding test Integration tests.

### CODE REVIEW

**Review by JingQianCloud — APPROVED** [review]
Looks good to me.

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java [hunk 1]
```diff
@@ -245,6 +253,15 @@ public void handleAdminRequest(NetworkRequest request) throws InterruptedExcepti
           requestTotalTimeHistogram = metrics.healthCheckRequestTotalTimeInMs;
           response = handleHealthCheckRequest(requestStream, adminRequest);
           break;
+        case BlobIndex:
+          metrics.blobIndexRequestQueueTimeInMs.update(requestQueueTime);
+          metrics.blobIndexRequestRate.mark();
+          processingTimeHistogram = metrics.blobIndexRequestQueueTimeInMs;
```
Comment (Arun-LinkedIn) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java:259]: Minor: Should this be `blobIndexRequestProcessingTimeInMs`

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java [hunk 2]
```diff
@@ -245,6 +253,15 @@ public void handleAdminRequest(NetworkRequest request) throws InterruptedExcepti
...
+          responseQueueTimeHistogram = metrics.blobIndexRequestQueueTimeInMs;
```
Comment (Arun-LinkedIn) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java:260]: Minor: Should this be `blobIndexResponseQueueTimeInMs`

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java [hunk 3]
```diff
@@ -503,10 +524,54 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
...
+      PartitionId partitionId = blobId.getPartition();
+      Store store = storeManager.getStore(partitionId);
+      if (store == null) {
+        logger.error("BlobId {}'s partition[{}] doesn't exist or stopped in this host", blobId, partitionId.getId());
+        return new AdminResponseWithContent(correlationId, clientId, ServerErrorCode.Replica_Unavailable, null);
+      }
+
+      Map<String, MessageInfo> messages = store.findAllMessageInfosForKey(blobId);
+      // Serialize with JSON
+      String json = objectMapper.writeValueAsString(messages);
```
Comment (Arun-LinkedIn) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java:559]: Just for my understanding. Would this return empty string if input is empty map?

Comment (justinlin-linkedin) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java:559]: it will return "{}" if the input the empty map.

**Pull request merged**